### ISSUE TITLE
Get "error creating disk" when creating vm

### ISSUE THREAD

**Issue by htj** [terraform-providers/terraform-provider-vsphere#525]
### Terraform Version

Terraform v0.11.7 
+ provider.vsphere (unversioned)

### vSphere Provider Version

vsphere plugin is git as of May 11th 2018.

I get the same behaviour on 1.4.1

### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```
provider "vsphere" {
    user           = "${var.vsphere_user}",
    password       = "${var.vsphere_password}",
    vsphere_server = "${var.vsphere_server}",
    allow_unverified_ssl = true # allow self-signed cert
}

resource "vsphere_virtual_machine" "test-vm" {
    resource_pool_id    = "ha-root-pool"

    name      = "test-vm"
    #folder    = "test-vm" # only supported on vcenter, not esxi
    num_cpus  = 1
    memory    = 512

    network_interface {
        network_id = "se-ndn-internal_services-01"
    }

    disk {
        label = "disk0"
        size = "15"
    }
```

### Debug Output

https://gist.github.com/htj/3ea9a220367c936942c4072c78f740e1

### Expected Behavior

Creating a VM.

What should have happened?

### Actual Behavior

```
vsphere_virtual_machine.test-vm: Creating...
  boot_retry_delay:                          "" => "10000"
  change_version:                            "" => "<computed>"
  cpu_limit:                                 "" => "-1"
  cpu_share_count:                           "" => "<computed>"
  cpu_share_level:                           "" => "normal"
  datastore_id:                              "" => "<computed>"
  default_ip_address:                        "" => "<computed>"
  disk.#:                                    "0" => "1"
  disk.0.attach:                             "" => "false"
  disk.0.datastore_id:                       "" => "<computed>"
  disk.0.device_address:                     "" => "<computed>"
  disk.0.disk_mode:                          "" => "persistent"
  disk.0.disk_sharing:                       "" => "sharingNone"
  disk.0.eagerly_scrub:                      "" => "false"
  disk.0.io_limit:                           "" => "-1"
  disk.0.io_reservation:                     "" => "0"
  disk.0.io_share_count:                     "" => "0"
  disk.0.io_share_level:                     "" => "normal"
  disk.0.keep_on_remove:                     "" => "false"
  disk.0.key:                                "" => "0"
  disk.0.label:                              "" => "disk0"
  disk.0.path:                               "" => "<computed>"
  disk.0.size:                               "" => "15"
  disk.0.thin_provisioned:                   "" => "true"
  disk.0.unit_number:                        "" => "0"
  disk.0.uuid:                               "" => "<computed>"
  disk.0.write_through:                      "" => "false"
  ept_rvi_mode:                              "" => "automatic"
  firmware:                                  "" => "bios"
  force_power_off:                           "" => "true"
  guest_id:                                  "" => "other-64"
  guest_ip_addresses.#:                      "" => "<computed>"
  host_system_id:                            "" => "<computed>"
  hv_mode:                                   "" => "hvAuto"
  imported:                                  "" => "<computed>"
  latency_sensitivity:                       "" => "normal"
  memory:                                    "" => "512"
  memory_limit:                              "" => "-1"
  memory_share_count:                        "" => "<computed>"
  memory_share_level:                        "" => "normal"
  migrate_wait_timeout:                      "" => "30"
  moid:                                      "" => "<computed>"
  name:                                      "" => "test-vm"
  network_interface.#:                       "0" => "1"
  network_interface.0.adapter_type:          "" => "vmxnet3"
  network_interface.0.bandwidth_limit:       "" => "-1"
  network_interface.0.bandwidth_reservation: "" => "0"
  network_interface.0.bandwidth_share_count: "" => "<computed>"
  network_interface.0.bandwidth_share_level: "" => "normal"
  network_interface.0.device_address:        "" => "<computed>"
  network_interface.0.key:                   "" => "<computed>"
  network_interface.0.mac_address:           "" => "<computed>"
  network_interface.0.network_id:            "" => "se-ndn-internal_services-01"
  num_cores_per_socket:                      "" => "1"
  num_cpus:                                  "" => "1"
  reboot_required:                           "" => "<computed>"
  resource_pool_id:                          "" => "ha-root-pool"
  run_tools_scripts_after_power_on:          "" => "true"
  run_tools_scripts_after_resume:            "" => "true"
  run_tools_scripts_before_guest_shutdown:   "" => "true"
  run_tools_scripts_before_guest_standby:    "" => "true"
  scsi_controller_count:                     "" => "1"
  scsi_type:                                 "" => "pvscsi"
  shutdown_wait_timeout:                     "" => "3"
  swap_placement_policy:                     "" => "inherit"
  uuid:                                      "" => "<computed>"
  vapp_transport.#:                          "" => "<computed>"
  vmware_tools_status:                       "" => "<computed>"
  vmx_path:                                  "" => "<computed>"
  wait_for_guest_net_routable:               "" => "true"
  wait_for_guest_net_timeout:                "" => "5"

Error: Error applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.test-vm: 1 error(s) occurred:

* vsphere_virtual_machine.test-vm: disk.0: error creating disk: ServerFaultCode: The object has already been deleted or has not been completely created

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

I get this error message no matter what permutation of creating a disk I've tried.

### Steps to Reproduce

`terraform apply`

### Important Factoids

ESXi 6.0.

Not vSphere. I suspect this is part of the reason, as ESXi is a lot more restrictive than vSphere.

**Comment by TT-JMay** [comment]
Any update on this? We are having the same exact issues.

**Comment by bwillif1** [comment]
I had this issue too, and was spinning my wheels trying to find a solution. I ended up getting it to work after I specified a variable to use for "datastore_id" under the disk section. I can see the output of your terraform state file shows:
**disk.0.datastore_id:                       "" => "<computed>"**

After I made the change, it displayed the actual datastore ID instead of saying "<computed". My VM started creating in vmware. Worth a try
